File: backend/pokemon-details/src/main/java/dev/thapelo/pokemon_details/stats/Stats.java
package dev.thapelo.pokemon_details.stats;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;

import java.util.Map;

@Getter
@Setter
@Entity
@Table(name = "stats")
@ToString
public class Stats {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private Long pokemonId;
    private String name;
    private Integer base_stat;
    private Integer effort;

    @JsonProperty("stat")
    private void extractName(Map<String, String> stat) {
        name = stat.get("name");
    }
}

File: frontend/src/app/pokemon/models/Stats.ts
export class Stats {
  id: Number;
  pokemon_id: Number;
  name: string;
  base_stat: Number;
  effort: Number;

  constructor(id: Number, pokemonId: Number, name: string, base_stat: Number, effort:Number) {
    this.id = id;
    this.pokemon_id = pokemonId;
    this.name = name;
    this.base_stat = base_stat;
    this.effort = effort;
  }
}
